lib/evaluate/intrinsics.cc

```diff
@@ -1618,40 +1617,34 @@ std::optional<SpecificCall> IntrinsicProcTable::Implementation::Probe(
        ++specIter) {
     // We only need to check the cases with distinct generic names.
```

Comment:
This comment may now be obsolete with the new approach.

Reply:
I think it still holds to justify the `if (const char *genericName{specIter->second->generic})` just below. 
My understanding is:
If `specIter->second->generic` is `nullptr`, then the specific name is also a generic name and the related generic that has already been attempted few lines before (and did not match if we got here). Hence there is no need to attempt the specific version that will fail because it is more restrictive than the generic version that already failed.